server/mongo_data_access.go

```diff
@@ -96,19 +154,45 @@ func (dal *mongoDataAccessLayer) Delete(id, resourceType string) error {
 		return convertMongoErr(err)
 	}
 
+	if dal.hasInterceptorsForVerbAndType("DELETE", resourceType) {
+		// Although this is a delete operation we need to get the resource first so we can
+		// run any interceptors on the resource before it's deleted.
+		resource, _ := dal.Get(id, resourceType)
+		dal.invokeInterceptors("DELETE", resourceType, resource)
+	}
+
 	collection := dal.Database.C(models.PluralizeLowerResourceName(resourceType))
 	return convertMongoErr(collection.RemoveId(bsonID.Hex()))
 }
 
 func (dal *mongoDataAccessLayer) ConditionalDelete(query search.Query) (count int, err error) {
+
+	resourceType := query.Resource
+
+	if dal.hasInterceptorsForVerbAndType("DELETE", resourceType) {
+		// get the resources that are about to be deleted
+		var bundle *models.Bundle
+		bundle, _ = dal.Search(url.URL{}, query) // the baseURL argument here does not matter
+
+		var resourceId string
+		for _, elem := range bundle.Entry {
+			resourceId = reflect.ValueOf(elem.Resource).Elem().FieldByName("Id").String()
+			err = dal.Delete(resourceId, resourceType)
+		}
+		count = int(*bundle.Total)
+		return count, convertMongoErr(err)
```

Comment:
This isn't exactly what I had in mind.  I was thinking more along the line of still using Mongo's `RemoveAll`, but passing in a query that just consists of the collected IDs (e.g., id is abc or id is xyz, etc.)  That would perform better and allow you to re-use much of the code below -- since you'd still be doing a query-based delete.  Does that make sense?